### DISCUSSION TITLE
crtBreakAlloc like functionality

### DISCUSSION THREAD

**Discussion by mtungekar** [mjansson/rpmalloc#267] [Ideas]
Are there any plans to support crtBreakAlloc like functionality in the future?

**Comment by mjansson** [comment]
The problem with this is the highly parallel nature of the allocator, there is no global order of the allocations as two threads can allocate blocks concurrently. Still, it could be useful for early initialization allocations which usually happen in a single thread linear order.

But it's probably easier for you to put it in place in the wrappings around rpmalloc, for example in the allocation tracker.

**Reply by mtungekar** [reply]
Yea that is what I was looking at few weeks ago. I will probably get back to it once I have time.. Thanks for the input.
